components/Chips/src/MDCChipCollectionViewFlowLayout.m

```diff
@@ -32,23 +32,24 @@ @implementation MDCChipCollectionViewFlowLayout
   NSMutableArray *customLayoutAttributes =
       [NSMutableArray arrayWithCapacity:layoutAttributes.count];
 
-  if (layoutAttributes.count > 0) {
-    UICollectionViewLayoutAttributes *attrs = [layoutAttributes[0] copy];
-    attrs.frame = CGRectLeftAlign(attrs.frame);
-    [customLayoutAttributes addObject:attrs];
-  }
-
-  for (NSUInteger i = 1; i < layoutAttributes.count; i++) {
-    UICollectionViewLayoutAttributes *attrs = [layoutAttributes[i] copy];
-    UICollectionViewLayoutAttributes *prevAttrs = customLayoutAttributes[i - 1];
-
-    if (CGRectGetMinY(prevAttrs.frame) == CGRectGetMinY(attrs.frame)) {
-      attrs.frame =
-          CGRectLeftAlignToRect(attrs.frame, prevAttrs.frame, self.minimumInteritemSpacing);
-    } else {
-      attrs.frame = CGRectLeftAlign(attrs.frame);
+  // Left-align cells
+  UICollectionViewLayoutAttributes *prevAttrs;
+  for (UICollectionViewLayoutAttributes *attrs in layoutAttributes) {
+    UICollectionViewLayoutAttributes *newAttrs = [attrs copy];
+    if (newAttrs.representedElementCategory == UICollectionElementCategoryCell) {
+      // If the attributes are for a cell that isn't on the same line as the previous cell, set the
+      // x origin of the frame to 0. Otherwise, align the frame to the right end of the previous
+      // frame (accounting for minimumInteritemSpacing).
+      if (!prevAttrs || (CGRectGetMinY(newAttrs.frame) != CGRectGetMinY(prevAttrs.frame))) {
```

Comment:
I would probably have a `BOOL isNewLine` local variable for `(CGRectGetMinY(newAttrs.frame) != CGRectGetMinY(prevAttrs.frame))` but the component is probably sufficient in explaining what's going on

Reply:
I like it! Done.